lib/java/com/google/android/material/imageview/ShapeableImageView.java

```diff
@@ -169,28 +169,25 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
       return;
     }
 
-    if (VERSION.SDK_INT > 19 && !isLayoutDirectionResolved()) {
-      return;
-    }
-
-    hasAdjustedPaddingAfterLayoutDirectionResolved = true;
-
-    // Update the super padding to be the combined `android:padding` and
-    // `app:contentPadding`, keeping with ShapeableImageView's internal padding contract:
-    if (VERSION.SDK_INT >= 21 && (isPaddingRelative() || isContentPaddingRelative())) {
-      setPaddingRelative(
-          super.getPaddingStart(),
-          super.getPaddingTop(),
-          super.getPaddingEnd(),
-          super.getPaddingBottom());
-      return;
+    if (VERSION.SDK_INT < 19 || isLayoutDirectionResolved()) {
+      hasAdjustedPaddingAfterLayoutDirectionResolved = true;
+
+      // Finally, update the super padding to be the combined `android:padding` and
+      //  `app:contentPadding`, in keeping with ShapeableImageView's crazy internal padding contract:
+      if (VERSION.SDK_INT >= 17 && (isPaddingRelative() || isContentPaddingRelative())) {
+        setPaddingRelative(
+            super.getPaddingStart(),
+            super.getPaddingTop(),
+            super.getPaddingEnd(),
+            super.getPaddingBottom());
+      } else {
+        setPadding(
+            super.getPaddingLeft(),
+            super.getPaddingTop(),
+            super.getPaddingRight(),
+            super.getPaddingBottom());
+      }
```

Comment:
When my previous PR #1871 was merged (outside of GitHub), this logic was refactored incorrectly and caused strange behavior on API 19. I've restored the original logic from #1871 along with the other logic changes/fixes in this PR.